lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -92,9 +92,10 @@ def fully_formatted_lines(failure_number, colorizer)
 
       private
 
-        def final_exception(exception)
-          if exception.cause
-            final_exception(exception.cause)
+        def final_exception(exception, previous=[])
+          if exception.cause && !previous.include?(exception.cause)
+            previous << exception.cause
+            final_exception(exception.cause, previous)
```

Comment:
It'll probably be slightly faster if you call `cause` only once:

``` ruby
def final_exception(exception, previous=[])
  cause = exception.cause
  return exception unless cause
  return exception if previous.include?(cause)

  previous << cause
  final_exception(cause, previous)
end
```

Not a merge blocker, though.